src/Sarif.Viewer.VisualStudio/Tags/SarifLocationTagger.cs

```diff
@@ -0,0 +1,573 @@
+            }
+
+            disposed = true;
+
+            if (disposing)
+            {
+                using (TagListLock.EnterReadLock())
+                {
+                    // Important note that we do not dispose or clear
+                    // the SARIF tag list as that would destroy the whole purpose of using
+                    // "persistent spans" which survive open and close of a document within
+                    // a VS session.
+                    foreach (var sarifTag in this.sarifTags)
+                    {
+                        sarifTag.PropertyChanged -= this.SarifTagPropertyChanged;
+                    }
+                }
+
+                using (SarifTaggersLock.EnterWriteLock())
+                {
+                    SarifTaggers.Remove(this);
+                }
+
+                this.batchUpdateLock.InnerLock.Dispose();
+            }
+        }
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        private void SarifTagPropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            if (sender is SarifLocationTag sarifTag)
+            {
+                using (this.Update())
+                {
+                    this.UpdateBatchSpan(sarifTag.DocumentPersistentSpan.Span);
+                }
+            }
+        }
+
+        private void UpdateBatchSpan(ITrackingSpan snapshotSpan)
+        {
+            // If there currently is a batch span, update it to include the biggest
+            // range of buffer affected so far.
+            if (this.batchUpdateSpan == null)
+            {
+                this.batchUpdateSpan = snapshotSpan;
+                return;
+            }
+
+            ITextSnapshot snapshot = this.textBuffer.CurrentSnapshot;
+
+            SnapshotSpan currentBatchSpan = this.batchUpdateSpan.GetSpan(snapshot);
+            SnapshotSpan currentUpdate = snapshotSpan.GetSpan(snapshot);
+
+            SnapshotPoint newStart = currentBatchSpan.Start.Position < currentUpdate.Start.Position ? currentBatchSpan.Start : currentUpdate.Start;
+            SnapshotPoint newEnd = currentBatchSpan.End.Position > currentUpdate.End.Position ? currentBatchSpan.End : currentUpdate.End;
+
+            this.batchUpdateSpan = snapshot.CreateTrackingSpan(new SnapshotSpan(newStart, newEnd), this.batchUpdateSpan.TrackingMode);
+        }
+
+        /// <inheritdoc/>
+        public void TextViewCreated(ITextView textView)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            if (!SdkUIUtilities.TryGetFileNameFromTextBuffer(textView.TextBuffer, out string textViewFileName))
+            {
+                return;
+            }
+
+            if (!textViewFileName.Equals(this.fileName, StringComparison.OrdinalIgnoreCase))
+            {
+                return;
+            }
+
+            new TextViewCaretListener(textView, this.sarifTags);
```

Comment:
>new TextViewCaretListener(textView, this.sarifTags) [](start = 12, length = 51)

We're creating an object to which nobody receives a reference. So we're calling this purely for its side effects. Why not just call a method that produces those side effects?

Reply:
Worth a comment to explain that the `textView` will hold a reference to the newly created object.

---
In reply to: [482264377](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r482264377) [](ancestors = 482264377)